src/components/attribute-picker/attribute-picker.js

```diff
@@ -0,0 +1,446 @@
+							aria-activedescendant="${this._dropdownIndex > -1 ? `attribute-dropdown-list-item-${this._dropdownIndex}` : ''}"
+							aria-autocomplete="list"
+							aria-expanded="${this._inputFocused}"
+							aria-haspopup="true"
+							aria-invalid="${ifDefined(ariaInvalid)}"
+							aria-label="${this.ariaLabel}"
+							aria-owns="attribute-dropdown-list"
+							aria-required=${ifDefined(ariaRequired)}
+							enterkeyhint="enter"
+							@blur="${this._onInputBlur}"
+							@focus="${this._onInputFocus}"
+							@keydown="${this._onInputKeydown}"
+							@input="${this._onInputTextChanged}"
+							type="text"
+							.value="${this._text}">
+
+						${(!isValid && !this._inputFocused) ? html`<div class="d2l-input-text-invalid-icon" @click="${this._handleInvalidIconClick}"></div>` : null}
+					</div>
+				`)}
+
+				<div class="d2l-attribute-picker-absolute-container">
+					<ul class="d2l-attribute-list"
+						role="listbox"
+						id="attribute-dropdown-list"
+						?hidden="${!this._inputFocused || !availableAttributes.length}">
+
+						${repeat(availableAttributes, (item) => item.name, (item, listIndex) => html`
+							<li id="attribute-dropdown-list-item-${listIndex}"
+								class="d2l-attribute-picker-li"
+								role="option"
+								aria-label="${this.localize('components:attributePicker:addValue', 'value', item.name)}"
+								aria-selected="${this._dropdownIndex === listIndex}"
+								.text="${item}"
+								.index=${listIndex}
+								@mousemove="${this._onListItemMouseMove}"
+								@mousedown="${this._onListItemTapped}">
+								${item.name}
+							</li>
+						`)}
+					</ul>
+				</div>
+			</div>
+
+			${!isValid ? html`
+				<d2l-tooltip for="d2l-attribute-picker-container" state="error" announced position="top" ?force-show=${this._inputFocused}>${this.invalidTooltipText || this.localize('components:attributePicker:minimumRequirement')}</d2l-tooltip>
+			` : null}
+		`;
+	}
+
+	updated(changedProperties) {
+		super.updated(changedProperties);
+		if (changedProperties.has('_activeAttributeIndex')) {
+			const selectedAttributes = this.shadowRoot.querySelectorAll('.d2l-attribute-picker-attribute');
+			if (this._activeAttributeIndex >= 0 && this._activeAttributeIndex < selectedAttributes.length) {
+				selectedAttributes[this._activeAttributeIndex].focus();
+			}
+		}
+
+		if (changedProperties.has('assignableAttributes')) {
+			this._dropdownIndex = -1;
+		}
+	}
+
+	// Returns true or false depending on if the attribute was successfully added. Fires the d2l-attribute-limit-reached event, if necessary.
+	async addAttribute(newAttribute) {
+		if (!newAttribute || this.attributeList.findIndex(attribute => attribute.name === newAttribute.name) >= 0) return false;
+
+		if (this.limit && this.attributeList.length >= this.limit) {
+			this.dispatchEvent(new CustomEvent('d2l-labs-attribute-picker-limit-reached', {
+				bubbles: true, composed: true, detail: { limit: this.limit }
+			}));
+			return false;
```

Comment:
There doesn't seem to be a tooltip stating the limit has been reached in this version. Did we intend to get rid of it?
![image](https://github.com/user-attachments/assets/4c0fa533-2608-4eba-af8e-31e21e4ce72a)

Reply:
That tooltip is found in the [foundation-components part](https://github.com/BrightspaceHypermediaComponents/foundation-components/blob/main/features/discover/d2l-discover-attribute-picker.js#L52) that implements this

Reply:
Should it be, or is this something that should belong to the picker itself for other consumers to benefit from?

Reply:
Hmm, that's a good question, we can talk more about it during stand-up on Monday, along with the point Dave just made about using the icon button instead of just the icon. I probably wouldn't hurt to have the tooltip as part of it.